src/client/pages/EventPage/index.tsx

```diff
@@ -209,6 +215,43 @@ class EventPage extends TabularPage<Props, EventPageState> {
     return (<SettingsTab {...this.props} />);
   }
 
+  /**
+   * Renders the event tab for editing the ebemt.
+   * @returns {Component} The edit tab
+   */
+  renderEditForm() {
+    const validator = createValidator();
+    const eventDate = new Date(this.props.event.closeTime);
+    const initialValues: Partial<EventFormData> = {
+      ...this.props.event,
+      closeTimeDay: eventDate.getDate(),
+      closeTimeMonth: eventDate.getMonth(),
+      closeTimeYear: eventDate.getFullYear(),
+      logo: undefined,
+    }
+
+    const editEvent = async (eventData: EventFormData) => {
+      try {
+        await editExistingEvent(this.props.event._id, eventData);
+        this.props.loadAllAdminEvents();
```

Comment:
I like the idea! I haven't tested it locally, but I can see a bug here -

1) Im editing /admin/events/hackxx
2) I edit eventAlias to `hackxx2020`

There's nothing here that updates the state of the current URL when the current eventalias is changed - should go into a 404 or an infinite loading state i think

adding a navigate `this.props.history.push to `eventData.alias` after adding the success alert should suffice here.

Reply:
yup I tested locally and it goes into an infinite loading state